### DISCUSSION TITLE
Extend Controller on v6.0 for Custom Action

### DISCUSSION THREAD

**Discussion by earthly-pixel** [Team-Tea-Time/laravel-forum#370] [Q&A] [answered]
Hello I am using extend controller in latest v5 configuration with custom namespace. With this, I can create custom actions like adding a score when creating a thread and so on. However, in the v6 configuration you can't use a custom namespace. Can you add that function to v6?

**Comment by Riari** [comment]
I'll look into this because I can't remember why I removed the namespace from the config. As a side note, could you do those things using events instead, or do you need to take in additional input with the existing forms/routes?

**Reply by earthly-pixel** [reply]
Yes, I need to add additional input with the current forms. I'll try to make a work around with the event for now. Thank you 😀

**Reply by Riari** [reply]
OK, that makes sense. I'll try to get back to you ASAP with an update as I know this is a bit of an annoying downgrade. I'll probably just reinstate the namespace option as that's very easy to do and obviously opens up lots of possibilities :)

**Reply by earthly-pixel** [reply]
Thank you for your hard work :)

**Reply by Riari** [reply, marked as answer]
I've taken a closer look and remember why I removed it now. It's because I refactored the routes to use the latest style of route definition, which involves proper namespace resolution instead of the old `Class@method` string approach. Obviously this has some benefits, but makes it impossible to tell the router to look in a different namespace.

I think the easiest alternative is probably just to override the specific forum routes that you want to extend the behaviour of - you can just point your route definitions to your existing controllers. Let me know if that doesn't work for some reason, but it should as long as you do the route definitions after the forum service provider!

**Reply by earthly-pixel** [reply]
Thanks, it works fine by overriding certain routes :)
